### DISCUSSION TITLE
'dxf_plane_fit.m'in user manual

### DISCUSSION THREAD

**Discussion by Micheal72506** [nicmenegoni/ROKA#2] [Q&A]
Dear sir, I am a graduate student studying in China. First of all, thank you for your selfless sharing, which inspired my thinking and made me gain a lot. I have a question to ask, that is, I don't see the script _'dxf_plane_fit.m'_ you mentioned in the user manual in the main program folder. If you can, can you provide more information about the script? I wish you all the best again!

**Comment by nicmenegoni** [comment]
Hello Micheal72506,
sorry, I forgot to upload the 'dxf_plane_fit.m' code because it is not strictly necessary for the ROKA algortihm.
Therefore, thank you for your notification!
I will upload in the next days, so stay tuned.
Cheers,
Niccolò

**Comment by nicmenegoni** [comment]
Dear Micheal72506,
I just upload the dxf_fitplane.m script.
Sorry for the late upload. 
Bests,
Niccolò

**Comment by Micheal72506** [comment]
Dear Niccolò Menegoni ,
Thank you for remembering my question and I express my great appreciation. During this time, I did some pre-processing work on ROKA. One thing I would like to confirm with you is whether the method you mentioned in your paper and the actual operation is manually extracting traces and fitting rectangular planes by CloudCompre software plug-in? If so, how long did it take to manually process the case in your paper where you fitted thousands of disc planes?
Bests,
WW

**Reply by nicmenegoni** [reply]
Hello,
the manual mapping could take some times. In general, to take 1000 maeasurments you have to spend at least one week mapping fractures. I noticed that the mapping ratio is usually from 30 to 100 fractures per hour. Obiouvlsy, it depends by your ability, the complexity of the rock slope, the resolution of the model and your mood.
After the manual mapping of the fracture traces (you can map also the fractures surfaces using the polyline tool and piccking a lot of points onto the fracture surface) you can operate in two ways:

1) save a DXF file containg all the fractures traces and surfaces mapped as polylines and then use the dxf_fitplane.m code to fit the circualr discs (best solution, becuase it prepares properly the data in XLSX);

2) fit the rectangular plane twith th 'fit plane' tool of CloudCompare, then export the plane info as CSV file using the bacth export tool of CloudCompare, and then adjust it calculating the radius as _**r=(sqrt(height^2 + width^2))/2**_ (using this formula you can obtain a radius size that is lilttle bit bigger than the real one). Then use change the headers of the columns as reported in the user manual and saved the spreadsheet as XSLX file.

**Comment by Micheal72506** [comment]
Hello Dear Sir, I apologize for not being able to reply to your email in time due to academic and internship reasons. Thank you for answering my questions in your busy schedule. Your answers were very constructive and beneficial to me. In your article titled Detection and geometric characterization of rock mass discontinuities using a 3D high-resolution digital outcrop model generated from RPAS imagery -Ormea rock slope, Italy, you mentioned the method of point cloud surface vegetation culling, and I would like to know where I can find the web version of this article. This is very important to me and I look forward to your reply.



            
            
                
                    
                            
                                
                            
                            
                                954135362
                            
                    
                        
                            
                                    ***@***.***
                            
                        
                
            
        Signature is customized by Netease Mail Master


            On 06/01/2021 20:58, Niccolò Menegoni wrote: 
Hello,
the manual mapping could take some times. In general, to take 1000 maeasurments you have to spend at least one week mapping fractures. I noticed that the mapping ratio is usually from 30 to 100 fractures per hour. Obiouvlsy, it depends by your ability, the complexity of the rock slope, the resolution of the model and your mood.
After the manual mapping of the fracture traces (you can map also the fractures surfaces using the polyline tool and piccking a lot of points onto the fracture surface) you can operate in two ways:


save a DXF file containg all the fractures traces and surfaces mapped as polylines and then use the dxf_fitplane.m code to fit the circualr discs (best solution, becuase it prepares properly the data in XLSX);


fit the rectangular plane twith th 'fit plane' tool of CloudCompare, then export the plane info as CSV file using the bacth export tool of CloudCompare, and then adjust it calculating the radius as r=(sqrt(height^2 + width^2))/2 (using this formula you can obtain a radius size that is lilttle bit bigger than the real one)



—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/nicmenegoni/ROKA/discussions/2#discussioncomment-810862",
"url": "https://github.com/nicmenegoni/ROKA/discussions/2#discussioncomment-810862",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Reply by nicmenegoni** [reply]
Hello Micheal72506, you can find the web version at [https://doi.org/10.1016/j.enggeo.2019.02.028], it is an open-access article. 
Cheers,
Niccolò

**Comment by Insar-wd** [comment]
Subject: Request for Assistance with ROKA Program
Dear Micheal72506,
I hope this message finds you well.
My name is [Wang Di], and I am a graduate student from China. I am currently conducting research related to the ROKA program. However, I have encountered some difficulties and would greatly appreciate any guidance or assistance you could provide.
If you are available to help, I would be truly grateful. You can reach me by phone at +86 182 6513 2919, or via email at wangdi9992024@163.com.
Thank you very much for your time and consideration. I look forward to hearing from you.
Warm regards,
